docs/README.md

```diff
@@ -1 +1,154 @@
+
+### Azure subscription requirements
+#### How to generate service principal
+Packer will authorize in Azure infrastructure via Service Principal. If you want to prepare image-generation CI, or use packer manually, you need to create your SP with full read-write permissions for selected Azure subscription.
+[Check more details](https://docs.microsoft.com/en-us/azure/active-directory/develop/howto-create-service-principal-portal)
+
+### Image generation process
+Virtual environment project using packer tool, and contain list of templates for several OS types, such as Windows 2016/2019, Ubuntu 16.04/18.04/20.04.
+Each image template contain set of scripts that will be executed on virtual machine.
+Packer process will initialize connection to Azure subscription via Azure Cli, and deploy required resources, such as temporary resource group, network interfaces and VM from "clean" image that specified in template.
+If deployment succeeded, build agent will connect with the deployed VM and begin to execute installation steps from selected template.
+Dependend of template connection may be estableshed from SSH or WinRM protocols, which mean following ports between build agent, and temporary VM must be opened for connection.
+If some of installation step from template fail, image generation will be aborted, and temporary VM will be terminated.
+In case if all installation steps are succeeded, temporary VM snapshot will be converted to VHD image and will be downloaded into selected Azure Storage Account, from which it may be deployed further.
+
+### Azure DevOps selfhosted pool requirements
+To connect with temporary VM packer will use WinRM, or SSH connections on public IP interfaces.
+If you use self-hosted pool located in some Azure subscription, please make sure that HTTPS/SSH ports are allowed for incoming/outgoing connections.
+In case in firewall restrictions forbid connections via public addresses, you may deploy private virtual network resources, and pass them as arguments to packer, so temporary virtual machines can use private connections inside vlan. 
+
+## Variables
+Each packer template include `variables` group that contains builder variables that used during image generation.
+Builder variables can be passed to packer via predefined environment variables, or as direct arguments, in case if packer started manually.
+
+### Common
+
+### Required builder variables
+- `client_id` - The application ID of the AAD Service Principal. Requires client_secret.
+- `object_id` - The object ID for the AAD SP. Optional, will be derived from the oAuth token if left empty.
+- `client_secret` - A password/secret registered for the AAD SP.
+- `subscription_id` - The subscription to use.
+- `tenant_id` - The Active Directory tenant identifier with which your client_id and subscription_id are associated. If not specified, tenant_id will be looked up using subscription_id.
```

Comment:
I think it makes sense to put all variables names here to backticks to highlight them